core/src/test/scala/integration/kafka/server/MultipleListenersWithSameSecurityProtocolTest.scala

```diff
@@ -57,9 +60,13 @@ class MultipleListenersWithSameSecurityProtocolTest extends ZooKeeperTestHarness
       // Ensure that we can support multiple listeners per security protocol and multiple security protocols
       props.put(KafkaConfig.ListenersProp, "SECURE_INTERNAL://localhost:0, INTERNAL://localhost:0, " +
         "SECURE_EXTERNAL://localhost:0, EXTERNAL://localhost:0")
-      props.put(KafkaConfig.ListenerSecurityProtocolMapProp, "INTERNAL:PLAINTEXT, SECURE_INTERNAL:SSL," +
-        "EXTERNAL:PLAINTEXT, SECURE_EXTERNAL:SSL")
+      props.put(KafkaConfig.ListenerSecurityProtocolMapProp, "INTERNAL:PLAINTEXT, SECURE_INTERNAL:SASL_SSL," +
+        "EXTERNAL:PLAINTEXT, SECURE_EXTERNAL:SASL_SSL")
       props.put(KafkaConfig.InterBrokerListenerNameProp, "INTERNAL")
+      props.put(KafkaConfig.ZkEnableSecureAclsProp, "true")
+      props.put(KafkaConfig.SaslMechanismInterBrokerProtocolProp, kafkaClientSaslMechanism)
+      props.put(KafkaConfig.SaslEnabledMechanismsProp, kafkaServerSaslMechanisms.mkString(","))
```

Comment:
To make the second test work with different login contexts for different mechanisms, you may also need to set:
```
    props.put(KafkaConfig.SaslKerberosServiceNameProp, "kafka")
```